pkg/services/sqlstore/migrations/ualert/tables.go

```diff
@@ -488,6 +488,9 @@ func addAlertImageMigrations(mg *migrator.Migrator) {
 	mg.AddMigration("support longer URLs in alert_image table", migrator.NewRawSQLMigration("").
 		Postgres("ALTER TABLE alert_image ALTER COLUMN url TYPE VARCHAR(2048);").
 		Mysql("ALTER TABLE alert_image MODIFY url VARCHAR(2048) NOT NULL;"))
+
+	mg.AddMigration("drop unique `token` index on alert_image", migrator.NewDropIndexMigration(imageTable, imageTable.Indices[0]))
+	mg.AddMigration("add unique index on `url` to alert_image", migrator.NewAddIndexMigration(imageTable, &migrator.Index{Type: migrator.UniqueIndex, Cols: []string{"url"}}))
```

Comment:
was URL already likely to be unique?

adding a unique index will fail if there are existing records that already break the uniqueness constraint. in what situations can existing images have the same URL? is this possible today? if so, do we need to clean out conflicting records before adding this?

if URL was guaranteed based on token before this change, like `.../path/to/some/<token>` it's safe. What do the URLs look like in practice?

Reply:
+ that will be broken when Grafana is configured to not upload anywhere and keep it only on disc (i.e. URL is empty)

Reply:
@alexweav thanks for pointing this out! The index indeed causes an error in the migrations.